controllers/downloads.js

```diff
@@ -179,6 +199,31 @@ var getRangeOfDays = function(request,reply,conditions,period) {
 
 }
 
+var handleBulkRange = function(rows, period, reply) {
+  var grouped = _.groupBy(rows, 'package'),
+    outputs = []
+
+  Object.keys(grouped).forEach(function(key) {
+    var dayCounts = _.map(grouped[key], function(r) {
+      return {
+        day: moment(r.day).format('YYYY-MM-DD'),
+        downloads: r.downloads
+      }
+    });
+
+    var output = {
+      downloads: dayCounts,
+      start: period.start,
+      end: period.end,
+      package: key
+    }
+
+    outputs.push(output)
+  })
+  
+  reply(outputs);
```

Comment:
This will force an API user to inspect the package field of each element of the array to work out what package this data is for. Maybe have the output be an object indexed by the package name? That would be much more convenient.